src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+        if isinstance(input_image, Image.Image):
+            input_image = [input_image]
+        if not isinstance(input_image, list) and not isinstance(input_image, tuple):
+            raise ValueError(f"Unsupported input image type: {type(input_image)}.")
+        num_images = len(input_image)
+
+        # latent checks
+        input_latent_stacked = False
+        if input_latent is not None:
+            if torch.is_tensor(input_latent):
+                if input_latent.ndim == 5:
+                    input_latent = [input_latent[i] for i in range(input_latent.shape[0])]
+                    input_latent_stacked = True
+                elif input_latent.ndim == 4:
+                    input_latent = [input_latent]
+                else:
+                    raise ValueError(f"Unsupported number of dimension in the input latent: {input_latent.ndim}.")
+            if isinstance(input_latent, list) and not isinstance(input_latent, tuple):
+                if not all(torch.is_tensor(k) for k in input_latent):
+                    raise ValueError("Input latent must be a torch.FloatTensor.")
+                if not all(
+                    k.dim() == 4 and k.shape[0] == ensemble_size and k.shape[1] == self.latent_space_size
+                    for k in input_latent
+                ):
+                    raise ValueError(
+                        f"Input latent must be 4-dimensional with shape [E,{self.latent_space_size},h,w], "
+                        f"where E is the requested ensemble_size."
+                    )
+                if len(input_latent) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and latents ({len(input_latent)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported latent type: {type(input_latent)}.")
+        if input_image_stacked ^ input_latent_stacked:
+            logger.warning("Different stacking of input images and latents might be a sign of undesired behavior.")
+
+        if generator is not None:
+            if isinstance(generator, torch.Generator):
+                generator = [generator]
+            if isinstance(generator, list) and not isinstance(generator, tuple):
+                if len(generator) == 1 and num_images > 1:
+                    generator = generator * num_images
+                if len(generator) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and generators ({len(generator)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+        # Prepare the empty text embedding. In the future, remove text modules completely
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
```

Comment:
Out of curiosity: If we want to remove it in the future anyway, why support it now?

Reply:
This is how it was implemented in the [community pipeline](https://github.com/huggingface/diffusers/blob/main/examples/community/marigold_depth_estimation.py). This tensor of shape=[1,2,1024] is not trivial, so we should either keep the language modules, or check it in explicitly. Given that we might want to support text conditioning in the future models, I'm inclined to do nothing about this bit, but let me know.

Reply:
Removed the comment